### PR TITLE
Initial release of the FivetranSyncTask, to manage your Fivetran connector sync process.

### PR DESCRIPTION
**Pull request by crw**
## Summary
Initial release of the FivetranSyncTask, to manage your [Fivetran](https://fivetran.com) connector sync process.

## Changes
Adds a `fivetran` folder (package) to `tasks`
Adds a FivetranSyncTask, starts and monitors a Fivetran connector job. 

## Importance
This is Fivetran's first effort at integrating our APIs directly into a scheduling tool. We are very much looking to see if this helps with more complicated use cases that involved loading data through Fivetran. 

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Tests have not actually been added yet. We would love to work with the team on the best way to test this new Task.

### CODE REVIEW

File: src/prefect/tasks/fivetran/__init__.py
Comment (jlowin) [inline src/prefect/tasks/fivetran/__init__.py:4]: Unless I'm mistaken, your task doesn't require any special dependencies, so it will actually be available automatically to every Prefect user! You could remove this `try/except` if you prefer.

Comment (crw) [reply src/prefect/tasks/fivetran/__init__.py:4]: I see! I was a little unclear on the purpose of this section of code. I'll pull that out.

**Comment by madkinsz** [comment]
Sweet! One of the benefits of using a `Task` subclass like this is that you can provide some helper methods to make the `run()` method easier to parse. I'd recommend splitting it up if you have the time so it's easier for users and future maintainers to understand.

**Comment by crw** [comment]
I completely agree, this could use a refactor. The focus with this initial release is on getting a minimal viable product in front of users to see if more effort is indicated. Let me know if you would like to see those changes before we go forward. There are a few other changes I would like to make (particular around the request.Session settings and API error retries) as an early v2 to this code, so I was already planning some improvements for the near future.

**Comment by madkinsz** [comment]
> I completely agree, this could use a refactor. The focus with this initial release is on getting a minimal viable product in front of users to see if more effort is indicated. Let me know if you would like to see those changes before we go forward. There are a few other changes I would like to make (particular around the request.Session settings and API error retries) as an early v2 to this code, so I was already planning some improvements for the near future.

That makes sense. That's more @jcrist's call than mine though.

File: src/prefect/tasks/fivetran/fivetran.py
Comment (cicdw) [inline src/prefect/tasks/fivetran/fivetran.py:100]: Because most exceptions inherit from the base `Exception` class, just to avoid a weird edge case we should probably choose a more specific exception here and down below - maybe a `ValueError` for now?

File: src/prefect/tasks/fivetran/fivetran.py
Comment (cicdw) [inline src/prefect/tasks/fivetran/fivetran.py:111]: We can reserve this enhancement for v2, but as an FYI note that you can publish "link artifacts" that will get rendered in the UI in a special way; this `URL_LOGS` could be a good candidate for that.  Example code here that publishes a link artifact: https://github.com/PrefectHQ/prefect/blob/master/src/prefect/tasks/prefect/flow_run.py#L191

**Review by cicdw — CHANGES_REQUESTED** [review]
This looks great - thanks for the quick turnaround @crw and @PubChimps !

I have one request on the raised exception type, but otherwise LGTM.  Once merged, we can make sure that this gets added to the API reference docs 👍

File: src/prefect/tasks/fivetran/fivetran.py
Comment (crw) [reply src/prefect/tasks/fivetran/fivetran.py:100]: Ha, I was reading through the [list of built-in exceptions](https://docs.python.org/3/library/exceptions.html) and nothing really seemed to fit. I'll look again, but yeah maybe just ValueError.

File: src/prefect/tasks/fivetran/fivetran.py
Comment (crw) [reply src/prefect/tasks/fivetran/fivetran.py:111]: Are there docs for this? I am a little confused how this works just from this code sample.

Comment (crw) [reply src/prefect/tasks/fivetran/fivetran.py:111]: I agree the existing implementation is a little hokey. I'm slightly concerned about them from the POV that Fivetran's UI URLs aren't necessarily a user-facing, supported feature in the way I am using them; they could change without notice.

Comment (cicdw) [reply src/prefect/tasks/fivetran/fivetran.py:111]: Gotcha, that makes sense; our artifacts API is also [very lightly documented](https://docs.prefect.io/api/latest/artifacts/artifacts.html) at the moment.  For a link artifact, the UI produces a clickable link in a convenient place on the task run / flow run pages for each run.

I'd vote to hold off on this enhancement for now given what you've described 👍

**Review by cicdw — APPROVED** [review]

File: src/prefect/tasks/fivetran/fivetran.py
Comment (cicdw) [reply src/prefect/tasks/fivetran/fivetran.py]: Yea, maybe we could use a [Prefect FAIL signal](https://docs.prefect.io/api/latest/engine/signals.html#fail) which simply indicates failure without having to pick an underlying exception.

Happy to merge without that, or happy to make that change myself post-merge, whatever you'd like

**Pull request merged**
